### PR TITLE
Enable Link PM for subscriptions

### LINKED ISSUE #2446: StripeLink: subscription support
Check Link implementation needs to be adjusted for subscriptions workflow (built-in subscriptions and subscription plugin) .

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #2446

## Changes proposed in this Pull Request:
Link payment method can be reused. For subscriptions, the payment method token should be saved in order to be re-used for subscription renewal.
This PR updates Link payment method to save the token.

## Testing instructions
**Scenario 1**
* Install [Woocommerce Subscriptions](https://woocommerce.com/products/woocommerce-subscriptions/)
* Create a new subscription order and pay it using Link
* Payment should be completed
* The payment of renewal for the subscription should be processed without errors.
* In order to trigger subscription renewals, you can follow the steps [ [here](https://woocommerce.com/document/testing-subscription-renewal-payments/) ]

**Scenario 2**
* Create a new subscription with 1 day trial version order and pay it using Link
* Payment should be completed
* The payment for the subscription should be processed without errors after the trial ends.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: woocommerce-gateway-stripe.php [hunk 1]
```diff
@@ -205,6 +205,7 @@ public function init() {
 				require_once dirname( __FILE__ ) . '/includes/admin/class-wc-stripe-upe-compatibility-controller.php';
 				require_once dirname( __FILE__ ) . '/includes/migrations/class-allowed-payment-request-button-types-update.php';
 				require_once dirname( __FILE__ ) . '/includes/class-wc-stripe-account.php';
+				require_once dirname( __FILE__ ) . '/includes/class-wc-stripe-link-payment-token.php';
```
Comment (kalessil) [inline woocommerce-gateway-stripe.php]: Can we relocate this include statement next to the SEPA token require statement?

**Review by kalessil — APPROVED** [review]
LGTM! I don't have a local test environment, hence would appreciate other reviewers to test the changes.

File: woocommerce-gateway-stripe.php (hunk 1 above)
Comment (dpaun1985) [reply woocommerce-gateway-stripe.php]: changed in [e010c7a](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2455/commits/e010c7af3141127f1cd9a7146ac986611708a664). thank you!

File: tests/phpunit/test-class-wc-stripe-upe-payment-method.php [hunk 2]
```diff
@@ -409,18 +424,27 @@ public function test_create_payment_token_for_user() {
...
-				$sepa_payment_method_mock = $this->array_to_object( self::MOCK_SEPA_PAYMENT_METHOD_TEMPLATE );
-				$token                    = $payment_method->create_payment_token_for_user( $user_id, $sepa_payment_method_mock );
-				$this->assertTrue( 'WC_Payment_Token_SEPA' === get_class( $token ) );
-				$this->assertEquals( $token->get_last4(), $sepa_payment_method_mock->sepa_debit->last4 );
-				$this->assertEquals( $token->get_token(), $sepa_payment_method_mock->id );
+			switch ( $payment_method_id ) {
+				case WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID:
+					$card_payment_method_mock = $this->array_to_object( self::MOCK_CARD_PAYMENT_METHOD_TEMPLATE );
+					$token                    = $payment_method->create_payment_token_for_user( $user_id, $card_payment_method_mock );
+					$this->assertTrue( 'WC_Payment_Token_CC' === get_class( $token ) );
```
Comment (kalessil) [inline tests/phpunit/test-class-wc-stripe-upe-payment-method.php:431]: And strict assertions, please =)

Comment (dpaun1985) [reply tests/phpunit/test-class-wc-stripe-upe-payment-method.php:431]: done. thx

**Review by reykjalin — APPROVED** [review]
Looks good and tests well. Great work! 👏

**Review by KarlisJ — APPROVED** [review]
LGTM! Code review only.

**Pull request merged**